File: sources/Dortware 1.0/club/dortware/client/event/impl/KeyboardEvent.java
package club.dortware.client.event.impl;

import club.dortware.client.event.Event;

public class KeyboardEvent extends Event {

    private final int key;

    public KeyboardEvent(int key) {
        this.key = key;
    }

    public int getKey() {
        return key;
    }
}

File: sources/Leaf devEnv/com/leafclient/leaf/event/client/KeyboardEvent.kt
package com.leafclient.leaf.event.client

import com.leafclient.leaf.utils.client.keyboard.Key
import fr.shyrogan.publisher4k.Cancellable

/**
 * Events related to the [KeyboardEvent]
 */
abstract class KeyboardEvent(val key: Key): Cancellable() {

    /**
     * Event invoked when a key is pressed
     */
    class Press(key: Key): KeyboardEvent(key)

    /**
     * Event invoked when a key is released
     */
    class Release(key: Key): KeyboardEvent(key)

}